### ISSUE TITLE
Floating point error in Hill climbing algorithm prevents the procedure from stopping.

### ISSUE THREAD

**Issue by karttikeya** [pgmpy/pgmpy#809]
Once the score gets low enough ( < e-10 ) sometimes the procedure gets stuck in a loop because of very slight change in score by an operation and again a perceived increase by reversing it or doing it again in case of flipping the edge.

**Comment by ankurankan** [comment]
@karttikeya Can you please share your code? It will be easier for us to reproduce it.

**Comment by karttikeya** [comment]
Sure. I was trying to construct a DAG for feature selection using IAMB algorithm. Following is the snippet: 

```python
import pandas as pd 
from pgmpy.models import BayesianModel
from pgmpy.estimators import BdeuScore
from pgmpy.estimators import HillClimbSearch
data = pd.read_csv('print.csv')
hc = HillClimbSearch(data, scoring_method=BicScore(data))      
best_model = hc.estimate()                                     
print best_model.edges()
```

I have modified the `HillClimbSearch.py` to print the operation executed and the score achieved. I have attached the corresponding output and also the input file Print.csv . 
[output.txt.txt](https://github.com/pgmpy/pgmpy/files/675281/output.txt.txt)

[Print.zip](https://github.com/pgmpy/pgmpy/files/675120/Print.zip)

**Comment by karttikeya** [comment]
I have submitted a simple patch for the problem; to break from the loop in case the score becomes too small.However, the patch shows a coveralls error which is to be expected due to the typical nature of the error and code. Kindly review.

**Comment by ankurankan** [comment]
Fixed in #811
